Tests/MongoSwiftTests/SpecTestRunner/Match.swift

```diff
@@ -10,34 +10,58 @@ internal protocol Matchable {
     /// This assumes `expected` is NOT a placeholder value (i.e. 42/"42"). Use `matches` if `expected` may be a
     /// placeholder.
     /// https://github.com/mongodb/specifications/tree/master/source/connection-monitoring-and-pooling/tests#spec-test-match-function
-    func contentMatches(expected: Any) -> Bool
```

Comment:
the new API allowed me to rewrite most of the `Matchable` stuff in a much better way, which was cool.